src/dom.js

```diff
@@ -6,21 +6,21 @@ let isDOMAvailable = typeof document === "object" && !!document.documentElement;
 let dom = {
     isDOMAvailable,
 
-    doc: isDOMAvailable ? document : null,
-    html: isDOMAvailable ? document.documentElement : null,
-    body: isDOMAvailable ? document.body : null,
+    doc: () => isDOMAvailable ? document : null,
+    html: () => isDOMAvailable ? document.documentElement : null,
+    body: () => isDOMAvailable ? document.body : null,
```

Comment:
Could you explain what is the reason for this change? At this point in the code, we already know whether `document` and `documentElement` exist or not, so it appears that we can store the DOM references immediately. The primary purpose of these three properties is to cache DOM object references thus avoiding repeated lookups in the prototype chains and execution of the corresponding getters in the future. Wrapping these into functions nullifies the effect of caching and makes these properties somewhat useless…

Reply:
Hi @Amphiluke,

Sorry I didn't not describe this part because it was committed a bit after I opened the PR.

The reason for this change is that storing the reference to the body/document breaks compatibility with hotwired/turbo. Basically this library handles page change itself by Ajax when you click an anchor link, so when you store a reference like you did, it refers to the previous page prior navigation. Doing this fix the issue.

Also getting read of these 3 methods and calling explicitly `document.*` everytime would be fine too

Reply:
Okay, got it, thanks.

> Also getting rid of these 3 methods and calling explicitly `document.*` everytime would be fine too

So, let’s maybe get rid of them indeed? Currently, they just add an extra-step of additional invocation without providing any benefits.